pkg/adapter/awssqssource/adapter.go

```diff
@@ -84,56 +83,49 @@ func NewAdapter(ctx context.Context, envAcc pkgadapter.EnvConfigAccessor, ceClie
 	}
 }
 
-const waitTimeoutSec = 20
-
 // Start implements adapter.Adapter.
 func (a *adapter) Start(stopCh <-chan struct{}) error {
 	url, err := a.queueLookup(a.arn.Resource)
 	if err != nil {
-		a.logger.Fatalw("Unable to find URL of SQS queue "+a.arn.Resource, zap.Error(err))
+		a.logger.Errorw("Unable to find URL of SQS queue "+a.arn.Resource, zap.Error(err))
+		return err
 	}
 
 	queueURL := *url.QueueUrl
 	a.logger.Infof("Listening to SQS queue at URL: %s", queueURL)
 
-	// Look for new messages every 5 seconds
-	ticker := time.NewTicker(5 * time.Second)
-	for range ticker.C {
-		msgs, err := a.getMessages(queueURL, waitTimeoutSec)
-		if err != nil {
-			a.logger.Errorw("Failed to get messages from SQS queue", "error", err)
-			continue
-		}
-
-		// Only push if there are messages on the queue
-		if len(msgs) < 1 {
-			continue
-		}
-
-		attributes, err := a.sqsClient.GetQueueAttributes(&sqs.GetQueueAttributesInput{
-			AttributeNames: []*string{aws.String("QueueArn")},
-			QueueUrl:       url.QueueUrl,
-		})
-		if err != nil {
-			a.logger.Errorw("Failed to get queue attributes", "error", err)
-			continue
-		}
-
-		err = a.sendSQSEvent(msgs[0], attributes.Attributes["QueueArn"])
-		if err != nil {
-			a.logger.Errorw("Failed to send event", "error", err)
-			continue
-		}
-
-		// Delete message from queue if we pushed successfully
-		err = a.deleteMessage(queueURL, msgs[0].ReceiptHandle)
-		if err != nil {
-			a.logger.Errorw("Failed to delete message from SQS queue", "error", err)
-			continue
+	backoff := common.NewBackoff()
+
+	err = backoff.Run(stopCh, func() func(context.Context) (bool, error) {
+		return func(ctx context.Context) (bool, error) {
```

Comment:
This one still uses the "complex" signature, is that on purpose?

Reply:
no, it isn't :see_no_evil: